src/lib/coda_base/ancestor.ml

```diff
@@ -35,7 +51,15 @@ module Prover : sig
     -> body_hash:State_body_hash.t
     -> unit
 
-  val prove : t -> source:State_hash.t -> ancestor:State_hash.t -> proof option
+  val verify_and_add :
+       t
+    -> Input.t
+    -> Output.t
+    -> ancestor_length:Coda_numbers.Length.t
+    -> Proof.t
+    -> unit Or_error.t
+
+  val prove : t -> Input.t -> (Output.t * Proof.t) option
```

Comment:
We don't actually ever want to use this module to generate the proofs. Live nodes will generate the proofs by consulting their transition frontier. Otherwise we'd be duplicating a lot of information in this datastructure.